SU2_CFD/src/numerics/flow/flow_diffusion.cpp

```diff
@@ -622,8 +599,24 @@ CNumerics::ResidualType<> CAvgGradInc_Flow::ComputeResidual(const CConfig* confi
                     dist_ij_2, nVar);
   }
 
+  /*--- Wall shear stress values (wall functions) ---*/
+
+  const int scale = max(1, (TauWall_i > 0.0)+(TauWall_j > 0.0));
+  Mean_TauWall = (max(TauWall_i,0.0) + max(TauWall_j,0.0)) / scale;
+
+  /*--- If using UQ methodology, set Reynolds Stress tensor and perform perturbation ---*/
+
+  if (using_uq){
+    ComputePerturbedRSM(nDim, Eig_Val_Comp, uq_permute, uq_delta_b, uq_urlx,
+                        Mean_GradPrimVar+1, Mean_PrimVar[nDim+2], Mean_Eddy_Viscosity,
+                        Mean_turb_ke, MeanPerturbedRSM);
+  }
+
   /*--- Get projected flux tensor (viscous residual) ---*/
-  SetStressTensor(Mean_PrimVar, Mean_GradPrimVar, Mean_turb_ke, Mean_Laminar_Viscosity, Mean_Eddy_Viscosity);
+  SetStressTensor(Mean_PrimVar, Mean_GradPrimVar, Mean_turb_ke,
+                  Mean_Laminar_Viscosity, Mean_Eddy_Viscosity);
+  if (config->GetQCR()) AddQCR(nDim, &Mean_GradPrimVar[1], tau);
+  if (Mean_TauWall > 0) AddTauWall(UnitNormal, Mean_TauWall);
```

Comment:
I've been looking into making the wall functions compatible with the vectorized classes.
I got confused, however, because it looks like the tau wall was not considered in the incompressible viscous numerics.
But your validation results looked good... So, either this makes them better, or it is not needed (here and in the other 2 viscous classes that use it).

Reply:
In #1120 @EduardoMolina also added the "tau wall logic" to the incompressible class but he changed the implementation in the 3 classes to only consider tau wall if only one of the two points has it.
Which means that for edges connecting two wall points tau wall is not considered... might be worth re-running some tests?

Reply:
I think in the case of RANS wall functions, the tauwall values that you mention are always overwritten by the tauwall computed at the end of CIncNSSolver::SetTauWall_WF(..). The function addtauwall is used only for the LES implementation.

Reply:
Tau wall is computed in that function as part of the preprocessing, to then be used in the numerics, I don't see any other place in the code where the computed tau wall can influence the solution.